sdk/ml/azure-ai-ml/azure/ai/ml/_schema/core/fields.py

```diff
@@ -109,17 +109,27 @@ def _serialize(self, value, attr, obj, **kwargs) -> typing.Optional[str]:
         if value is None:
             return None
         self._validate(value)
-        return super(LocalPathField, self)._serialize(value, attr, obj, **kwargs)
+        # always dump path as absolute path as base_path will be dropped after serialization
+        return super(LocalPathField, self)._serialize(
+            self.__resolve_path(value).as_posix(),
+            attr,
+            obj,
+            **kwargs
+        )
+
+    def __resolve_path(self, value):
```

Comment:
why  __ not _ here?

Reply:
Add some docstrings

Reply:
updated. Want to make it private in the beginning, but seems not necessary